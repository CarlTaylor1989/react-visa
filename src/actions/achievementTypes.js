export const SET_COMPLETED_ACHIEVEMENT = 'SET_COMPLETED_ACHIEVEMENT';
export const RESTORE_ACHIEVEMENTS = 'RESTORE_ACHIEVEMENTS';

export const UPDATE_PRODUCT_ACHIEVEMENTS = 'UPDATE_PRODUCT_ACHIEVEMENTS';
export const UPDATE_PROPOSAL_ACHIEVEMENTS = 'UPDATE_PROPOSAL_ACHIEVEMENTS';
export const UPDATE_SCORED_ACHIEVEMENTS = 'UPDATE_SCORED_ACHIEVEMENTS';
export const UPDATE_SETTINGS_ACHIEVEMENTS = 'UPDATE_SETTINGS_ACHIEVEMENTS';
export const UPDATE_TIMED_ACHIEVEMENTS = 'UPDATE_TIMED_ACHIEVEMENTS';

export const RESET_CHANGED_ACHIEVEMENT = 'RESET_CHANGED_ACHIEVEMENT';
export const RESET_ACHIEVEMENTS = 'RESET_ACHIEVEMENTS';
