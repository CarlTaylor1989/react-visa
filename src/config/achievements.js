export const groups = {
  challenges: 'challenges',
  products: 'products',
  proposals: 'proposals',
  scored: 'scored',
  settings: 'settings',
  timed: 'timed',
};

export default [
  {
    id: 'ac1',
    score: 5000,
    hidden: false,
    group: 'scored',
    max: 1,
    available: false
  }, {
    id: 'ac2',
    score: 5000,
    hidden: false,
    group: 'challenges',
    max: 1,
    available: false
  }, {
    id: 'ac3',
    score: 10000,
    hidden: false,
    group: 'products',
    max: 1,
    available: true
  }, {
    id: 'ac4',
    score: 50000,
    hidden: false,
    group: 'products',
    max: 5,
    available: false
  }, {
    id: 'ac5',
    score: 150000,
    hidden: false,
    group: 'products',
    max: 15,
    available: false
  }, {
    id: 'ac6',
    score: 250000,
    hidden: false,
    group: 'products',
    max: 25,
    available: false
  }, {
    id: 'ac7',
    score: 100000,
    hidden: true,
    group: 'timed',
    max: 5,
    available: true
  }, {
    id: 'ac8',
    score: 1,
    hidden: true,
    attempts: 5,
    group: 'proposals',
    available: true
  }, {
    id: 'ac9',
    score: 1111,
    hidden: false,
    group: 'scored',
    max: 1000000,
    available: true
  }, {
    id: 'ac10',
    score: 33333,
    hidden: false,
    group: 'scored',
    max: 3000000,
    available: true
  }, {
    id: 'ac11',
    score: 5555,
    hidden: true,
    group: 'scored',
    max: 5000000,
    available: true
  }, {
    id: 'ac12',
    score: 10101,
    hidden: true,
    group: 'scored',
    max: 10000000,
    available: true
  }, {
    id: 'ac13',
    score: 50000,
    hidden: false,
    group: 'proposals',
    available: false
  }, {
    id: 'ac14',
    score: 250000,
    hidden: false,
    group: 'proposals',
    max: 8,
    level: 1,
    available: false
  }, {
    id: 'ac15',
    score: 500000,
    hidden: false,
    group: 'proposals',
    max: 8,
    level: 2,
    available: false
  }, {
    id: 'ac16',
    score: 750000,
    hidden: false,
    group: 'proposals',
    max: 8,
    level: 3,
    available: false
  }, {
    id: 'ac17',
    score: 1000000,
    hidden: true,
    group: 'proposals',
    max: 24,
    allClients: true,
    available: false
  }, {
    id: 'ac18',
    score: 25000,
    hidden: true,
    attempts: 0,
    group: 'proposals',
    available: true
  }, {
    id: 'ac19',
    score: 5000,
    hidden: true,
    group: 'challenges',
    progress: 3,
    available: true
  }, {
    id: 'ac20',
    score: 1000000,
    hidden: true,
    group: 'challenges',
    max: 100,
    perfect: true,
    available: false
  }, {
    id: 'ac21',
    score: 5000,
    hidden: true,
    group: 'settings',
    max: 1,
    available: false
  }
];
