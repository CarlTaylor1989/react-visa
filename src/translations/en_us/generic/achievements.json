{
  "ac1": {
    "title": "OFF THE MARK",
    "description": "SCORE YOUR FIRST POINTS"
  },
  "ac2": {
    "title": "ONE DOWN",
    "description": "COMPLETE YOUR FIRST CHALLENGE"
  },
  "ac3": {
    "title": "FLYING START",
    "description": "COMPLETE YOUR FIRST PRODUCT"
  },
  "ac4": {
    "title": "TAKE 5",
    "description": "COMPLETE 5 PRODUCTS (IN ANY ORDER)"
  },
  "ac5": {
    "title": "SMARTY PANTS",
    "description": "COMPLETE 15 PRODUCTS (IN ANY ORDER)"
  },
  "ac6": {
    "title": "CLEVER CLOGGS",
    "description": "COMPLETE 25 PRODUCTS (IN ANY ORDER)"
  },
  "ac7": {
    "title": "HOOKED",
    "description": "PLAY THE GAME FIVE DAYS IN A ROW"
  },
  "ac8": {
    "title": "YOU'll GET IT NEXT TIME!",
    "description": "FAIL A CLIENT PROPOSAL FIVE TIMES"
  },
  "ac9": {
    "title": "MILLIONNAIRE",
    "description": "SCORE 1 MILLION POINTS"
  },
  "ac10": {
    "title": "M-M-MILLIONAIRE",
    "description": "SCORE 3 MILLION POINTS"
  },
  "ac11": {
    "title": "YOU MIGHT BE PLAYING TOO MUCH",
    "description": "SCORE 5 MILLION POINTS"
  },
  "ac12": {
    "title": "THIS SHOULDN'T BE POSSIBLE",
    "description": "SCORE 10 MILLION POINTS"
  },
  "ac13": {
    "title": "FIRST OF MANY",
    "description": "COMPLETE YOUR FIRST CLIENT PROPOSAL"
  },
  "ac14": {
    "title": "CLIENTS LOVE YOU (LEVEL 1)",
    "description": "COMPLETE ALL LEVEL 1 CLIENT PROPOSALS"
  },
  "ac15": {
    "title": "CLIENTS LOVE YOU (LEVEL 2)",
    "description": "COMPLETE ALL LEVEL 2 CLIENT PROPOSALS"
  },
  "ac16": {
    "title": "CLIENTS LOVE YOU (LEVEL 3)",
    "description": "COMPLETE ALL LEVEL 3 CLIENT PROPOSALS"
  },
  "ac17": {
    "title": "EPIC WIN!",
    "description": "COMPLETE ALL CLIENT PROPOSALS"
  },
  "ac18": {
    "title": "FIRST TIME'S A CHARM",
    "description": "COMPLETE A CLIENT PROPOSAL ON THE FIRST ATTEMPT"
  },
  "ac19": {
    "title": "FELL AT THE LAST HURDLE",
    "description": "FAIL A CHALLENGE WHEN ON A BONUS STREAK OF 3"
  },
  "ac20": {
    "title": "PERFECTIONIST",
    "description": "COMPLETE ALL CHALLENGES WITH A PERFECT SCORE"
  },
  "ac21": {
    "title": "FINE TUNA",
    "description": "ACCESS THE SETTINGS PANEL"
  }
}
