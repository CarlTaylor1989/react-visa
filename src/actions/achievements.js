import * as types from './achievementTypes';

export function setCompletedAchievement(achievement) {
  return {
    type: types.SET_COMPLETED_ACHIEVEMENT,
    achievement
  };
}

export function restoreAchievements(completed, lastCompleted) {
  return {
    type: types.RESTORE_ACHIEVEMENTS,
    completed,
    lastCompleted
  };
}

export function updateProductAchievements() {
  return {
    type: types.UPDATE_PRODUCT_ACHIEVEMENTS
  };
}

export function updateProposalAchievements() {
  return {
    type: types.UPDATE_PROPOSAL_ACHIEVEMENTS
  };
}

export function updateScoredAchievements() {
  return {
    type: types.UPDATE_SCORED_ACHIEVEMENTS
  };
}

export function updateSettingsAchievements() {
  return {
    type: types.UPDATE_SETTINGS_ACHIEVEMENTS
  };
}

export function updateTimedAchievements() {
  return {
    type: types.UPDATE_TIMED_ACHIEVEMENTS
  };
}

export function resetChangedAchievement() {
  return {
    type: types.RESET_CHANGED_ACHIEVEMENT
  };
}

export function resetAchievements() {
  return {
    type: types.RESET_ACHIEVEMENTS
  };
}
