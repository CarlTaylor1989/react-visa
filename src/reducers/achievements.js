import { createReducer } from 'redux-create-reducer';
import { cloneDeep } from 'lodash';
import * as types from '../actions/achievementTypes';

export const initialState = {
  completed: [],
  lastCompleted: '',
  completedThisSession: false,
  achievementChanged: false
};

export const achievementsData = createReducer(cloneDeep(initialState), {
  [types.SET_COMPLETED_ACHIEVEMENT](state, action) {
    const exists = state.completed.includes(action.achievement);
    return {
      ...state,
      completed: exists ? state.completed : [
        ...state.completed,
        action.achievement
      ],
      achievementChanged: exists ? state.achievementChanged : true,
      lastCompleted: exists ? state.lastCompleted : action.achievement,
      completedThisSession: exists ? state.completedThisSession : true
    };
  },
  [types.RESTORE_ACHIEVEMENTS](state, action) {
    return {
      ...state,
      completed: [
        ...action.completed
      ],
      lastCompleted: action.lastCompleted
    };
  },
  [types.RESET_CHANGED_ACHIEVEMENT](state) {
    return {
      ...state,
      achievementChanged: initialState.achievementChanged
    };
  },
  [types.RESET_ACHIEVEMENTS]() {
    return cloneDeep(initialState);
  }
});

export default { achievementsData };
