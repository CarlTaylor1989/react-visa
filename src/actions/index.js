import { push } from 'connected-react-router';
import * as AchievementActions from './achievements';
import * as AudioActions from './audio';
import * as ClientActions from './client';
import * as DiagnosticActions from './diagnostic';
import * as GenericActions from './generic';
import * as HintsActions from './hints';
import * as MapActions from './map';
import * as PromptActions from './prompts';
import * as ScoreActions from './score';
import * as ScormActions from './scorm';
import * as SettingsActions from './settings';
import * as StreakActions from './streak';
import * as TutorialActions from './tutorial';

const ActionCreators = Object.assign(
  {},
  AchievementActions,
  AudioActions,
  ClientActions,
  DiagnosticActions,
  GenericActions,
  HintsActions,
  MapActions,
  PromptActions,
  ScoreActions,
  ScormActions,
  SettingsActions,
  StreakActions,
  TutorialActions,
  { push }
);

export default ActionCreators;
