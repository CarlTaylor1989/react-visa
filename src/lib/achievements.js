import _ from 'lodash';
import achievements, { groups } from '../config/achievements';

export const lib = {};

/**
 * Gets achievements sorted by score descending.
 * @return {array}
 */
lib.sortedAchievements = () => _.orderBy(achievements, 'score', 'desc');

/**
 * Gets the percentage complete for an achievement based
 * on how many tasks have been completed.
 * @param {number} completed Total completed
 * @param {number} total     Total tasks
 * @return {number}
 */
lib.getPercentageComplete = (completed, total) => {
  let percent;
  if (completed > 0) {
    const rounded = _.round((completed / total) * 100);
    percent = _.clamp(rounded, 0, 100);
  } else {
    percent = 0;
  }
  return percent;
};

/**
 * Returns achievements with their completion data.
 * @param {object} data The data required to get the achievement status
 * @return {object}
 */
export const getUserAchievementData = (data) => {
  let sortedAchv = lib.sortedAchievements();
  sortedAchv = _.filter(sortedAchv, ach => ach.available !== false);
  return sortedAchv.map((achievement) => {
    const complete = data.completed.includes(achievement.id);
    let completion = {
      complete,
      completedAmount: 0,
      completedPercent: complete ? 100 : 0,
      totalAmount: achievement.max || 1
    };
    switch (achievement.group) {
      case groups.scored: {
        const completedAmount = (data.total > completion.totalAmount)
          ? completion.totalAmount
          : data.total;
        completion = {
          ...completion,
          completedAmount,
          completedPercent: lib.getPercentageComplete(completedAmount, completion.totalAmount)
        };
        break;
      }
      case groups.challenges: {
        if (achievement.max) {
          if (achievement.perfect) {
            completion.completedAmount = complete ? achievement.max : data.noOfPerfectChallenges;
            completion.completedPercent = lib.getPercentageComplete(
              data.noOfPerfectChallenges,
              achievement.max
            );
          } else {
            completion.completedAmount = complete ? achievement.max : data.noOfCompletedChallenges;
            completion.completedPercent = lib.getPercentageComplete(
              data.noOfCompletedChallenges,
              achievement.max
            );
          }
        } else if (achievement.progress && complete) {
          completion.completedAmount = 1;
        }
        break;
      }
      case groups.products: {
        completion = {
          ...completion,
          completedAmount: complete ? achievement.max : data.noOfCompletedProducts,
          completedPercent: lib.getPercentageComplete(data.noOfCompletedProducts, achievement.max)
        };
        break;
      }
      case groups.timed: {
        completion.completedAmount = complete ? achievement.max : data.noOfConsecutiveDays;
        if (!complete) {
          completion.completedPercent = lib.getPercentageComplete(
            data.noOfConsecutiveDays,
            achievement.max
          );
        }
        break;
      }
      case groups.settings: {
        if (completion.complete) {
          completion = {
            ...completion,
            completedAmount: 1
          };
        }
        break;
      }
      case groups.proposals:
        if (achievement.level || achievement.allClients) {
          const completed = data.clients.filter(cl => (
            cl.completed
            && (achievement.allClients || (achievement.level && cl.level === achievement.level))
          ));
          completion.completedAmount = completed.length;
          completion.completedPercent = lib.getPercentageComplete(
            completion.completedAmount,
            achievement.max
          );
        } else {
          completion.completedAmount = completion.complete ? 1 : 0;
        }
        break;
      default: {
        completion.completedAmount = completion.complete ? 1 : 0;
        break;
      }
    }

    return {
      ...achievement,
      completion
    };
  });
};
