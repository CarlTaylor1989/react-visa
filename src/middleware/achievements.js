import _ from 'lodash';
import * as achievActions from '../actions/achievements';
import * as achievTypes from '../actions/achievementTypes';
import * as mapTypes from '../actions/mapTypes';
import * as promptActions from '../actions/prompts';
import * as scoreActions from '../actions/score';
import achievements, { groups } from '../config/achievements';
import {
  CHALLENGE_FAILED_STATUS,
  CHALLENGE_COMPLETION_STATUS,
  CHALLENGE_PERFECT_STATUS
} from '../config/constants';

const groupedAchievs = _.groupBy(achievements, ac => ac.group);

export const lib = {};
lib.getGroupedAchievements = group => groupedAchievs[group];

lib.dispatchActions = (dispatch, id, score) => {
  dispatch(achievActions.setCompletedAchievement(id));
  dispatch(promptActions.addAchievementPrompt(id));
  dispatch(scoreActions.increaseScore(score));
  dispatch(scoreActions.increaseAchievementsUnlocked());
};

export const challengeAchievements = ({ dispatch, getState }) => next => (action) => {
  next(action);

  if (action.type === mapTypes.UPDATE_CHALLENGE_STATUS) {
    const state = getState();
    const completedAchievements = state.achievementsData.completed;
    const completedChallenges = state.mapData.completedChallenges;
    const streakProgres = state.streakData.progress;
    const challengeAchiev = lib.getGroupedAchievements(groups.challenges);

    challengeAchiev.forEach((ac) => {
      if (!completedAchievements.includes(ac.id)) {
        if (ac.available && ((ac.max && completedChallenges.length === ac.max
          && (action.status === CHALLENGE_COMPLETION_STATUS
            || action.status === CHALLENGE_PERFECT_STATUS))
          || (streakProgres === ac.progress && action.status === CHALLENGE_FAILED_STATUS))) {
          lib.dispatchActions(dispatch, ac.id, ac.score);
        }
      }
    });
  }
};

export const productAchievements = ({ dispatch, getState }) => next => (action) => {
  next(action);

  if (action.type === mapTypes.ADD_COMPLETED_PRODUCT) {
    const state = getState();
    const competedProducts = state.mapData.completedProducts;
    const completedAchievements = state.achievementsData.completed;
    const productAchiev = lib.getGroupedAchievements(groups.products);

    productAchiev.forEach((ac) => {
      if (ac.available && competedProducts.length === ac.max
          && !completedAchievements.includes(ac.id)) {
        lib.dispatchActions(dispatch, ac.id, ac.score);
      }
    });
  }
};

export const settingsAchievement = ({ dispatch, getState }) => next => (action) => {
  next(action);

  if (action.type === achievTypes.UPDATE_SETTINGS_ACHIEVEMENTS) {
    const completedAchievements = getState().achievementsData.completed;
    const settingsAchiev = lib.getGroupedAchievements(groups.settings);

    settingsAchiev.forEach((ac) => {
      if (ac.available && !completedAchievements.includes(ac.id)) {
        lib.dispatchActions(dispatch, ac.id, ac.score);
      }
    });
  }
};

export const timedAchievement = ({ dispatch, getState }) => next => (action) => {
  next(action);

  if (action.type === achievTypes.UPDATE_TIMED_ACHIEVEMENTS) {
    const state = getState();
    const completedAchievements = state.achievementsData.completed;
    const consecutive = state.genericData.consecutive;
    const timedAchiev = lib.getGroupedAchievements(groups.timed);

    timedAchiev.forEach((ac) => {
      if (ac.available && !completedAchievements.includes(ac.id) && consecutive === ac.max) {
        lib.dispatchActions(dispatch, ac.id, ac.score);
      }
    });
  }
};

export const scoredAchievements = ({ dispatch, getState }) => next => (action) => {
  next(action);

  // Do we have a scoring action which sets the score?
  if (action.score) {
    const state = getState();
    const completedAchievements = state.achievementsData.completed;
    const totalScore = state.scoreData.total;
    const scoredAchiev = lib.getGroupedAchievements(groups.scored);

    scoredAchiev.forEach((ac) => {
      if (ac.available && !completedAchievements.includes(ac.id) && ac.max <= totalScore) {
        lib.dispatchActions(dispatch, ac.id, ac.score);
      }
    });
  }
};

export default [
  challengeAchievements,
  productAchievements,
  settingsAchievement,
  timedAchievement,
  scoredAchievements
];
