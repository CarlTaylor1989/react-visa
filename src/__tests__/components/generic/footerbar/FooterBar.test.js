/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  FooterBar,
  mapStateToProps,
  mapDispatchToProps
} from '../../../../components/generic/footerbar/FooterBar';
import * as actions from '../../../../actions/achievements';
import * as types from '../../../../actions/achievementTypes';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    changeRegion: jest.fn(),
    settingsBtnEnabled: true,
    setPlayAudio: jest.fn(),
    updateSettingsAchievements: jest.fn()
  };
});

test('renders the FooterBar component with an enabled settings button', () => {
  const wrapper = shallow(<FooterBar {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the FooterBar component with a disabled settings button', () => {
  mockProps.settingsBtnEnabled = false;
  const wrapper = shallow(<FooterBar {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('method onCloseSettings()', () => {
  const wrapper = shallow(<FooterBar {...mockProps} />);
  wrapper.instance().onCloseSettings();
  expect(wrapper.instance().state.displaySettings).toBeFalsy();
});

test('method onDisplaySettings()', () => {
  const wrapper = shallow(<FooterBar {...mockProps} />);
  wrapper.instance().onDisplaySettings();
  expect(wrapper.instance().state.displaySettings).toBeTruthy();
});

test('method onChangeRegion()', () => {
  // setup
  const wrapper = shallow(<FooterBar {...mockProps} />);
  wrapper.instance().onCloseSettings = jest.fn();

  // function under test
  wrapper.instance().onChangeRegion();

  // expectations
  expect(wrapper.instance().onCloseSettings).toHaveBeenCalled();
  expect(mockProps.changeRegion).toHaveBeenCalled();
});

test('method onHoverBtn()', () => {
  const wrapper = shallow(<FooterBar {...mockProps} />);
  wrapper.instance().onHoverBtn();
  expect(mockProps.setPlayAudio).toHaveBeenCalledWith(expect.any(String));
});

test('should mapStateToProps', () => {
  const state = {
    genericData: {
      settingsBtnEnabled: true
    }
  };
  const expected = {
    ...state.genericData
  };
  const props = mapStateToProps(state);
  expect(props).toEqual({
    ...expected
  });
});

test('should mapDispatchToProps', () => {
  const dispatch = jest.fn();
  jest.spyOn(actions, 'updateSettingsAchievements').mockReturnValue('updateSettingsAchievements()');
  const props = mapDispatchToProps(dispatch);
  props.updateSettingsAchievements();
  expect(dispatch).toHaveBeenCalledWith({
    type: types.UPDATE_SETTINGS_ACHIEVEMENTS
  });
});
