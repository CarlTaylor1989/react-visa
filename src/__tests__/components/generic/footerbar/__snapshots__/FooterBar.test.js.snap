// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FooterBar component with a disabled settings button 1`] = `
<div
  className="footerBar"
>
  <Connect(Hints) />
  <button
    className="settingsBtn disabled"
    disabled={true}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchEnd={[Function]}
    type="button"
  >
    <span
      className="icon"
    />
    <Translate
      tag="span"
      value="generic.settingsBtn"
    />
  </button>
  <Connect(SettingsPopup)
    changeRegion={[Function]}
    closePopup={[Function]}
    exitPopup={[MockFunction]}
    showPopup={false}
  />
</div>
`;

exports[`renders the FooterBar component with an enabled settings button 1`] = `
<div
  className="footerBar"
>
  <Connect(Hints) />
  <button
    className="settingsBtn "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchEnd={[Function]}
    type="button"
  >
    <span
      className="icon"
    />
    <Translate
      tag="span"
      value="generic.settingsBtn"
    />
  </button>
  <Connect(SettingsPopup)
    changeRegion={[Function]}
    closePopup={[Function]}
    exitPopup={[MockFunction]}
    showPopup={false}
  />
</div>
`;
